File: NDOTestApp/NDOTravelExpenses-Step 6 VB/BusinessClasses/Expense.vb
Imports NDO
Imports System


<NDOPersistent()> _
Public MustInherit Class Expense
    ' Properties
    Public MustOverride ReadOnly Property Amount() As Decimal

    Public Property [Date]() As DateTime
        Get
            Return _date
        End Get
        Set(ByVal value As DateTime)
            _date = value
        End Set
    End Property

    Public MustOverride ReadOnly Property Text() As String
    ' Fields
    Private _date As DateTime
End Class

File: Tutorial/NDOTestApp/NDOTravelExpenses-Step 6/BusinessClasses/Expense.cs
using System;
using NDO;
namespace BusinessClasses
{
    [NDOPersistent]
    public abstract partial class Expense
    {
        DateTime date;
        public DateTime Date
        {
            get { return date; }
            set { date = value; }
        }
        public abstract string Text
        {
            get;
        }
        public abstract decimal Amount
        {
            get;
        }
        public Expense()
        {
        }
    }
}
